rfcs/active/0000-badge.md

```diff
@@ -0,0 +1,362 @@
+| `text` | `string` | `null` | Content inside the ribbon |
+| `color` | `string` | `null` | Custom color for the ribbon |
+| `placement` | `'start' \| 'end'` | `'end'` | Placement of the ribbon (follows text direction) |
+
+### CSS Custom Properties
+
+The badge component will expose the following CSS custom properties for theming:
+
+```css
+:host {
+  /* Size and spacing */
+  --oui-badge-font-size: 0.75em;
+  --oui-badge-font-weight: 700;
+  --oui-badge-line-height: 1;
+  --oui-badge-padding-x: 0.65em;
+  --oui-badge-padding-y: 0.35em;
+  --oui-badge-border-radius: 0.25rem;
+  --oui-badge-pill-border-radius: 50rem; /* Used when shape="pill" */
+  --oui-badge-dot-size: 0.5rem;
+  --oui-badge-z-index: 10;
+  --oui-badge-offset-x: 0;
+  --oui-badge-offset-y: 0;
+
+  /* Color tokens - should use design system global tokens */
+  --oui-badge-color-primary-bg: var(--oui-color-primary, #0066cc);
+  --oui-badge-color-primary-text: var(--oui-color-primary-contrast, #ffffff);
+  --oui-badge-color-secondary-bg: var(--oui-color-secondary, #6c757d);
+  --oui-badge-color-secondary-text: var(--oui-color-secondary-contrast, #ffffff);
+  --oui-badge-color-success-bg: var(--oui-color-success, #198754);
+  --oui-badge-color-success-text: var(--oui-color-success-contrast, #ffffff);
+  --oui-badge-color-danger-bg: var(--oui-color-danger, #dc3545);
+  --oui-badge-color-danger-text: var(--oui-color-danger-contrast, #ffffff);
+  --oui-badge-color-warning-bg: var(--oui-color-warning, #ffc107);
+  --oui-badge-color-warning-text: var(--oui-color-warning-contrast, #000000);
+  --oui-badge-color-info-bg: var(--oui-color-info, #0dcaf0);
+  --oui-badge-color-info-text: var(--oui-color-info-contrast, #000000);
+
+  /* Status indicators */
+  --oui-badge-color-status-success: var(--oui-color-success, #52c41a);
+  --oui-badge-color-status-processing: var(--oui-color-info, #1890ff);
+  --oui-badge-color-status-default: var(--oui-color-secondary, #d9d9d9);
+  --oui-badge-color-status-error: var(--oui-color-danger, #ff4d4f);
+  --oui-badge-color-status-warning: var(--oui-color-warning, #faad14);
+
+  /* Ribbon specific */
+  --oui-badge-ribbon-height: 1.5rem;
+  --oui-badge-ribbon-padding-x: 0.5rem;
+  --oui-badge-ribbon-corner-size: 0.5rem;
+}
+```
+
+## Behavior
+
+### Sizing and Positioning
+
+Badges should automatically scale to match the size of the parent element when used inline (e.g., with headings or buttons). This will be achieved through using relative font sizes (`em` units).
+
+When the `position` attribute is used, the badge will be absolutely positioned relative to its parent container, which should have `position: relative` applied.
+
+### Accessibility Considerations
+
+1. **Color contrast**: All badge color variants must meet WCAG 2.1 AA contrast requirements.
+2. **Screen readers**:
+   - When a badge contains only a number (e.g., "4" for notifications), an `aria-label` should be provided to give context (e.g., "4 unread messages").
```

Comment:
what would be the role of `oui-badge`? If it's `generic`, naming is [explicitly prohibited](https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited) in WAI-ARIA 1.2.

In the example provided, a single number, there's often an associated icon and they are in a button together. Buttons are nameable, and if the icon has an alternative, like 'Messages', the total accessible name for the button would be something like 'Messages, 4'. Including the word 'unread' would probably only make sense if the icon conveyed 'unread' somehow. Otherwise, '4 messages' alone would create equity between visual and programmatic.